starport/services/network/network.go

```diff
@@ -2,273 +2,78 @@ package network
-		o.keyringBackend = keyringBackend
-	}
-}
-
-// InitializationGenesisURL provides a genesis url for the initial genesis of the chain blockchain
-func InitializationGenesisURL(genesisURL string) InitOption {
-	return func(o *initOptions) {
-		o.genesisURL = genesisURL
-	}
-}
-
-// AccountRegistry returns the account registry used by the network builder
-func (b Builder) AccountRegistry() cosmosaccount.Registry {
-	return b.cosmos.AccountRegistry
-}
+type LaunchInfo = networkchain.Launch
 
-// Blockchain initializes a blockchain from source and options.
-func (b *Builder) Blockchain(ctx context.Context, source SourceOption, options ...InitOption) (*Blockchain, error) {
-	var o initOptions
-	for _, apply := range options {
-		apply(&o)
-	}
-	source(&o)
+// LaunchInfo fetches the chain launch from Starport Network by launch id.
+func (n Network) LaunchInfo(ctx context.Context, id uint64) (LaunchInfo, error) {
+	n.ev.Send(events.New(events.StatusOngoing, "Fetching chain information"))
 
-	var (
-		genesisURL  = o.genesisURL
-		home        = o.homePath
-		url         = o.url
-		ref         = o.ref
-		hash        = o.hash
-		chainID     string
-		genesisHash string
-	)
-
-	// if a launch id is provided, chain information are fetched from Starport Network
-	if o.launchID > 0 {
-		b.ev.Send(events.New(events.StatusOngoing, "Fetching chain information"))
-		chainLaunch, err := b.fetchChainLaunch(ctx, o.launchID)
-		if err != nil {
-			return nil, err
-		}
-		b.ev.Send(events.New(events.StatusOngoing, "Chain information fetched"))
-		url = chainLaunch.SourceURL
-		hash = chainLaunch.SourceHash
-		chainID = chainLaunch.GenesisChainID
-
-		// Check if custom genesis URL is provided
-		if customGenesisURL := chainLaunch.InitialGenesis.GetGenesisURL(); customGenesisURL != nil {
-			genesisURL = customGenesisURL.Url
-			genesisHash = customGenesisURL.Hash
-		}
-
-		// If no custom home is provided, a default home determined from the launch ID is used
-		if home == "" {
-			home, err = ChainHome(o.launchID)
-			if err != nil {
-				return nil, err
-			}
-		}
-	}
-
-	b.ev.Send(events.New(events.StatusOngoing, "Fetching the source code"))
-	path, hash, err := b.fetchSource(ctx, url, ref, hash)
-	if err != nil {
-		return nil, err
-	}
-	b.ev.Send(events.New(events.StatusDone, "Source code fetched"))
-
-	bc := &Blockchain{
-		appPath:     path,
-		url:         url,
-		hash:        hash,
-		builder:     b,
-		genesisURL:  genesisURL,
-		genesisHash: genesisHash,
-	}
-	return bc, bc.setup(chainID, home, o.keyringBackend)
-}
-
-// fetchChainLaunch fetches the chain launch from Starport Network from a launch id
-func (b *Builder) fetchChainLaunch(ctx context.Context, launchID uint64) (launchtypes.Chain, error) {
-	res, err := launchtypes.NewQueryClient(b.cosmos.Context).Chain(ctx, &launchtypes.QueryGetChainRequest{
-		LaunchID: launchID,
+	res, err := launchtypes.NewQueryClient(n.cosmos.Context).Chain(ctx, &launchtypes.QueryGetChainRequest{
+		LaunchID: id,
```

Comment:
can we move these spn calls for a custom package, for we can abstract the spn types?

Reply:
we had a pkg/spn before that does this, to only interact with SPN. but it became too much, an unnecessary abstraction where we had to repeat the same funcs between two packages. I don't see a need for now, what do you think?

Reply:
Maybe we can just group the different queries like `LaunchInfo()` (it seems there is only this one for now in this PR) under a `query.go`

For messages, for now we only have one method for a message so I would say it's fine. In the future, if we implement a command for requesting a genesis account and a genesis validator individually maybe it would make sense to group some message sending method in a `msg.go` without creating a package

Reply:
let's do separate them to different files as needed, in the future PRs.